controllers/eventsource/resource.go

```diff
@@ -240,42 +252,25 @@ func buildDeployment(args *AdaptorArgs, eventBus *eventbusv1alpha1.EventBus) (*a
 				},
 			},
 		})
-		volumeMounts = append(volumeMounts, corev1.VolumeMount{Name: "auth-volume", MountPath: common.EventBusAuthFileMountPath})
+		volumeMounts = append(volumeMounts, corev1.VolumeMount{
+			Name:      "auth-volume",
+			MountPath: common.EventBusAuthFileMountPath,
+		})
 	}
-	deploymentSpec.Template.Spec.Volumes = volumes
-	deploymentSpec.Template.Spec.Containers[0].VolumeMounts = volumeMounts
 
-	envs := deploymentSpec.Template.Spec.Containers[0].Env
-	envs = append(envs, envVars...)
-	deploymentSpec.Template.Spec.Containers[0].Env = envs
+	// secrets
+	volSecrets, volSecretMounts := common.VolumesFromSecretsOrConfigMaps(common.SecretKeySelectorType, secretObjs...)
```

Comment:
When the eventBus type is kafka, we need can use the `common.VolumesFromSecretsOrConfigMaps` function to attach the tls and sasl secrets. They need to be attached at the same time as the secrets in the sensor/eventsource so that any secrets with the same name will be deduplicated.